### PR TITLE
chore: bump up pulsar dep

### LINKED ISSUE #6436: Switch back to plusar main branch
resolve #6417

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

change back to pulsar crate

## Checklist

~~- [ ] I have written necessary rustdoc comments~~
~~- [ ] I have added necessary unit tests and integration tests~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

resolve #6436

### CODE REVIEW

**Comment by tabVersion** [comment]
I have noticed the latest version is `4.2.0` but it requires exactly `uuid 1.2.1`, which causes dep conflict. I have tried loosening the constraint but it will not pass the tests.

File: Cargo.lock [hunk 1]
```diff
@@ -207,6 +217,36 @@ dependencies = [
  "event-listener",
 ]
 
+[[package]]
+name = "async-native-tls"
+version = "0.3.3"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "9e9e7a929bd34c68a82d58a4de7f86fffdaf97fb2af850162a7bb19dd7269b33"
+dependencies = [
+ "async-std",
```
Comment (BugenZhao) [inline Cargo.lock:226]: 😄

File: src/connector/Cargo.toml [hunk 2]
```diff
@@ -37,7 +37,7 @@ nexmark = { path = "../utils/nexmark", features = ["serde"] }
 num-traits = "0.2"
 paste = "1"
 prost = "0.11"
-pulsar = { version = "5", default-features = false, features = ["tokio-runtime"], rev = "7fab6a9", git = "https://github.com/skyzh/pulsar-rs" }
+pulsar = "4.1.3"
```
Comment (BugenZhao) [inline src/connector/Cargo.toml:40]: I guess you need to disable the default features.

Comment (tabVersion) [reply src/connector/Cargo.toml:40]: seem true. The release notes claim that they are compatible with tokio and async-std

File: src/workspace-hack/Cargo.toml [hunk 3]
```diff
@@ -146,13 +147,14 @@ phf_shared = { version = "0.11", features = ["std", "uncased"] }
 postgres-types = { git = "https://github.com/madsim-rs/rust-postgres.git", rev = "87ca1dc", default-features = false, features = ["chrono-04", "derive", "postgres-derive", "with-chrono-0_4"] }
 proc-macro2 = { version = "1", features = ["proc-macro", "span-locations"] }
 prometheus = { version = "0.13", features = ["libc", "process", "procfs", "protobuf"] }
-prost = { version = "0.11", features = ["no-recursion-limit", "prost-derive", "std"] }
+prost-93f6ce9d446188ac = { package = "prost", version = "0.10", features = ["prost-derive", "std"] }
+prost-a6292c17cd707f01 = { package = "prost", version = "0.11", features = ["no-recursion-limit", "prost-derive", "std"] }
```
Comment (BugenZhao) [inline src/workspace-hack/Cargo.toml:151]: Is this okay? cc @skyzh

**Comment by tabVersion** [comment]
@huangjw806 can you help trigger CI of [demo](https://github.com/risingwavelabs/risingwave-demo/tree/main/twitter-pulsar) based on this branch? we don't have pulsar CI on main

**Comment by neverchanje** [comment]
> @huangjw806 can you help trigger CI of [demo](https://github.com/risingwavelabs/risingwave-demo/tree/main/twitter-pulsar) based on this branch? we don't have pulsar CI on main

demo will only run against a fixed version of RisingWave and I upgrade it from time to time. It does not alway follow the main.And I don’t think it’s necessary as well.

As for the CI for Pulsar, we’ll need a more thorough plan for how to test sources other than Kafka. @lmatz The testing for Kinesis also needs to be considered.

**Comment by skyzh** [comment]
We should fork pulsar and loosen its constraints.

File: src/workspace-hack/Cargo.toml (hunk 3 above)
Comment (skyzh) [reply src/workspace-hack/Cargo.toml:151]: Then this PR didn't upgrade successfully to the latest version :(

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**